### ISSUE TITLE
setDistinct(Distinct.of(0)) is throwing an AlgoliaHttpException

### ISSUE THREAD

**Issue by pixelastic** [algolia/algoliasearch-client-java#343]
The following code seems to be throwing an exception:

```
Query q = new Query();
    q.setDistinct(Distinct.of(0));
    q.setNumericFilters(Lists.newArrayList("id=" + id));
SearchResult<MyRecord> results = myIndex.search(q);
```

I tried the same behavior with the Ruby client and here is the expected behavior:

- `false`/`0` should disable the distinct feature and return all results
- `true`/`1` should enable the distinct feature and return only one result
- `2`, `3`, etc should limit the number of result to that specific value
- `-1`, `99999`, `0.5`, `foo` will still be sent to the API, but the API will return an error

**Comment by ElPicador** [comment]
Yes, there was a bug in the serialization of the query for this parameter. The fix is pushed, I'll release the version `2.11.2` (I love palindromes) as soon as possible

**Comment by ElPicador** [comment]
Release 2.11.2 done!

Thanks for the bug report
